pkg/reconciler/pipelinerun/resources/pipelinerunresolution_test.go

```diff
@@ -3554,3 +3554,83 @@ func TestGetRunName(t *testing.T) {
 		})
 	}
 }
+
+func TestIsMatrixed(t *testing.T) {
+	pr := v1beta1.PipelineRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "pipelinerun",
+		},
+	}
+	getTask := func(ctx context.Context, name string) (v1beta1.TaskObject, error) { return task, nil }
+	getTaskRun := func(name string) (*v1beta1.TaskRun, error) { return nil, nil }
```

Comment:
Are these methods executed during the test? If they're not, maybe you could pass `nil` so it panics if they're unexpectedly called, or `return nil, errors.New("unimplemented")` so they fail a bit more gracefully.

Having `ResolvePipelineRunTask` take a bunch of methods has always felt like a pretty bad smell, and part of that is making these tests include all this copied boilerplate.